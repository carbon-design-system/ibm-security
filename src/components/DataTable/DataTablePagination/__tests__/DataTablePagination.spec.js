/**
 * @file Data table and pagination tests.
 * @copyright IBM Security 2019
 */

import { shallow } from 'enzyme';
import React from 'react';
import { DataTablePagination } from '../../';

import {
  headers,
  isSelectable,
  isSortable,
  page,
  pageSize,
  pageSizes,
  rows,
  render,
} from '../../_mocks_';

describe('DataTablePagination', () => {
  let dataTablePagination;
  let dataTablePaginationInstance;

  beforeEach(() => {
    dataTablePagination = shallow(
      <DataTablePagination
        rows={rows}
        headers={headers}
        isSelectable={isSelectable}
        isSortable={isSortable}
        page={page}
        pageSize={pageSize}
        pageSizes={pageSizes}
        render={render}
      />
    );

    dataTablePaginationInstance = dataTablePagination.instance();
  });

  describe('Rendering', () => {
    it('renders correctly', () => {
      expect(dataTablePagination).toMatchSnapshot();
    });

    it("renders the HTML of the node's subtree", () => {
      expect(dataTablePagination.render()).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('validates the state change for `paginationChange`', () => {
      const updateState = state => state * 2;

      const pageValue = updateState(page);
      const pageSizeValue = updateState(pageSize);

      dataTablePaginationInstance.paginationChange({
        page: pageValue,
        pageSize: pageSizeValue,
      });

      const {
        page: updatedPage,
        pageSize: updatedPageSize,
      } = dataTablePagination.state();

      expect(updatedPage).toEqual(pageValue);
      expect(updatedPageSize).toEqual(pageSizeValue);
    });

    it('validates data range for `paginateData`', () => {
      expect(dataTablePaginationInstance.paginateRows(rows).length).toEqual(
        pageSize
      );
    });
  });
});
