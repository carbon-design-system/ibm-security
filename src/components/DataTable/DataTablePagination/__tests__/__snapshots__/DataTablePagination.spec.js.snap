// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTablePagination Rendering renders correctly 1`] = `
<div
  className="security--data-table-pagination"
>
  <DataTable
    filterRows={[Function]}
    headers={
      Array [
        Object {
          "header": "Last sighted",
          "key": "lastSighted",
        },
        Object {
          "header": "Name",
          "key": "name",
        },
        Object {
          "header": "Type",
          "key": "type",
        },
        Object {
          "header": "Attack phase",
          "key": "attackPhase",
        },
        Object {
          "header": "Method",
          "key": "method",
        },
        Object {
          "header": "Missing",
          "key": "missing",
        },
      ]
    }
    isSelectable={false}
    isSortable={false}
    locale="en"
    missingDataCharacter="–"
    overflowMenuOnHover={true}
    rows={
      Array [
        Object {
          "attackPhase": "Initial Compromise",
          "id": "7c34f4507156b35",
          "lastSighted": "11/28/2018",
          "method": "Phishing",
          "name": "IBM X-Force",
          "type": "IPv6",
        },
        Object {
          "attackPhase": "Establish Foothold",
          "id": "38347816a51b0c5a8",
          "lastSighted": "06/28/2017",
          "method": "Deploy backdoor",
          "name": "AppX",
          "type": "Hash",
        },
        Object {
          "attackPhase": "Escalate Privileges",
          "id": "1d6116a2d1d03b4",
          "lastSighted": "05/30/2018",
          "method": "Credential dumping",
          "name": "Bigfix",
          "type": "URL",
        },
        Object {
          "attackPhase": "Cross-Site Scripting",
          "id": "c52b0419981221765",
          "lastSighted": "12/12/2018",
          "method": "Malicious Code",
          "name": "Am I Affected",
          "type": "Hash",
        },
        Object {
          "attackPhase": "SQL Injection Attack",
          "id": "697626e6f703131d",
          "lastSighted": "11/11/2018",
          "method": "SQL Vulnerabilities",
          "name": "QRadar",
          "type": "Hash",
        },
      ]
    }
    size="normal"
    sortRow={[Function]}
    translateWithId={[Function]}
  />
  <Pagination
    backwardText="Backward"
    disabled={false}
    forwardText="Forward"
    isLastPage={false}
    itemRangeText={[Function]}
    itemText={[Function]}
    itemsPerPageText="Items per page"
    onChange={[Function]}
    page={1}
    pageInputDisabled={false}
    pageNumberText="Page Number"
    pageRangeText={[Function]}
    pageSize={5}
    pageSizes={
      Array [
        5,
        10,
        20,
      ]
    }
    pageText={[Function]}
    pagesUnknown={false}
    totalItems={11}
  />
</div>
`;

exports[`DataTablePagination Rendering renders the HTML of the node's subtree 1`] = `
<div
  class="security--data-table-pagination"
>
  <div
    class="security--data-table"
  >
    <div
      class="bx--data-table-container"
    >
      <div
        class="security--scroll-gradient security--scroll-gradient--none security--scroll-gradient--x security--data-table__table-wrapper"
        role="presentation"
      >
        <div
          aria-hidden="true"
          class="security--scroll-gradient__before"
          role="presentation"
          style="background-image:linear-gradient(-90deg, rgba(0,0,0,0), var(--cds-ui-01, #262626) 90%)"
        />
        <div
          class="security--scroll-gradient__content"
        >
          <div
            class="bx--data-table-content"
          >
            <table
              class="bx--data-table bx--data-table--no-border"
            >
              <thead>
                <tr>
                  <th
                    scope="col"
                  >
                    <div
                      class="bx--table-header-label"
                    >
                      Last sighted
                    </div>
                  </th>
                  <th
                    scope="col"
                  >
                    <div
                      class="bx--table-header-label"
                    >
                      Name
                    </div>
                  </th>
                  <th
                    scope="col"
                  >
                    <div
                      class="bx--table-header-label"
                    >
                      Type
                    </div>
                  </th>
                  <th
                    scope="col"
                  >
                    <div
                      class="bx--table-header-label"
                    >
                      Attack phase
                    </div>
                  </th>
                  <th
                    scope="col"
                  >
                    <div
                      class="bx--table-header-label"
                    >
                      Method
                    </div>
                  </th>
                  <th
                    scope="col"
                  >
                    <div
                      class="bx--table-header-label"
                    >
                      Missing
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                aria-live="polite"
              >
                <tr>
                  <td>
                    11/28/2018
                  </td>
                  <td>
                    IBM X-Force
                  </td>
                  <td>
                    IPv6
                  </td>
                  <td>
                    Initial Compromise
                  </td>
                  <td>
                    Phishing
                  </td>
                  <td>
                    –
                  </td>
                </tr>
                <tr>
                  <td>
                    06/28/2017
                  </td>
                  <td>
                    AppX
                  </td>
                  <td>
                    Hash
                  </td>
                  <td>
                    Establish Foothold
                  </td>
                  <td>
                    Deploy backdoor
                  </td>
                  <td>
                    –
                  </td>
                </tr>
                <tr>
                  <td>
                    05/30/2018
                  </td>
                  <td>
                    Bigfix
                  </td>
                  <td>
                    URL
                  </td>
                  <td>
                    Escalate Privileges
                  </td>
                  <td>
                    Credential dumping
                  </td>
                  <td>
                    –
                  </td>
                </tr>
                <tr>
                  <td>
                    12/12/2018
                  </td>
                  <td>
                    Am I Affected
                  </td>
                  <td>
                    Hash
                  </td>
                  <td>
                    Cross-Site Scripting
                  </td>
                  <td>
                    Malicious Code
                  </td>
                  <td>
                    –
                  </td>
                </tr>
                <tr>
                  <td>
                    11/11/2018
                  </td>
                  <td>
                    QRadar
                  </td>
                  <td>
                    Hash
                  </td>
                  <td>
                    SQL Injection Attack
                  </td>
                  <td>
                    SQL Vulnerabilities
                  </td>
                  <td>
                    –
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="security--scroll-gradient__after"
          role="presentation"
          style="background-image:linear-gradient(-90deg, var(--cds-ui-01, #262626) 10%, rgba(0,0,0,0))"
        />
      </div>
    </div>
  </div>
  <div
    class="bx--pagination"
  >
    <div
      class="bx--pagination__left"
    >
      <label
        class="bx--pagination__text"
        for="bx-pagination-select-1"
        id="bx-pagination-select-1-count-label"
      >
        Items per page
      </label>
      <div
        class="bx--form-item"
      >
        <div
          class="bx--select bx--select--inline bx--select__item-count"
        >
          <div
            class="bx--select-input--inline__wrapper"
          >
            <div
              class="bx--select-input__wrapper"
            >
              <select
                class="bx--select-input"
                id="bx-pagination-select-1"
              >
                <option
                  class="bx--select-option"
                  selected=""
                  value="5"
                >
                  5
                </option>
                <option
                  class="bx--select-option"
                  value="10"
                >
                  10
                </option>
                <option
                  class="bx--select-option"
                  value="20"
                >
                  20
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="bx--select__arrow"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <span
        class="bx--pagination__text bx--pagination__items-count"
      >
        1-5 of 11 items
      </span>
    </div>
    <div
      class="bx--pagination__right"
    >
      <div
        class="bx--form-item"
      >
        <div
          class="bx--select bx--select--inline bx--select__page-number"
        >
          <label
            class="bx--label bx--visually-hidden"
            for="bx-pagination-select-1-right"
          >
            Page number, of 3 pages
          </label>
          <div
            class="bx--select-input--inline__wrapper"
          >
            <div
              class="bx--select-input__wrapper"
            >
              <select
                class="bx--select-input"
                id="bx-pagination-select-1-right"
              >
                <option
                  class="bx--select-option"
                  selected=""
                  value="1"
                >
                  1
                </option>
                <option
                  class="bx--select-option"
                  value="2"
                >
                  2
                </option>
                <option
                  class="bx--select-option"
                  value="3"
                >
                  3
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="bx--select__arrow"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <span
        class="bx--pagination__text"
      >
        1 of 3 pages
      </span>
      <div
        class="bx--pagination__control-buttons"
      >
        <button
          class="bx--pagination__button bx--pagination__button--backward bx--pagination__button--no-index bx--btn bx--btn--ghost bx--btn--disabled bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-center"
          disabled=""
          tabindex="0"
          type="button"
        >
          <div
            class="bx--assistive-text"
          >
            Backward
          </div>
          <svg
            aria-hidden="true"
            aria-label="Backward"
            class="bx--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 24L10 16 20 8z"
            />
          </svg>
        </button>
        <button
          class="bx--pagination__button bx--pagination__button--forward bx--btn bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-end"
          tabindex="0"
          type="button"
        >
          <div
            class="bx--assistive-text"
          >
            Forward
          </div>
          <svg
            aria-hidden="true"
            aria-label="Forward"
            class="bx--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8L22 16 12 24z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
